components/tidb_query/src/expr_util/string.rs

```diff
@@ -0,0 +1,74 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+
+use crate::codec::data_type::*;
+use base64;
+
+// see https://dev.mysql.com/doc/refman/5.7/en/string-functions.html#function_to-base64
+// mysql base64 doc: A newline is added after each 76 characters of encoded output
+pub const BASE64_LINE_WRAP_LENGTH: usize = 76;
+
+// mysql base64 doc: Each 3 bytes of the input data are encoded using 4 characters.
+pub const BASE64_INPUT_CHUNK_LENGTH: usize = 3;
+pub const BASE64_ENCODED_CHUNK_LENGTH: usize = 4;
+pub const BASE64_LINE_WRAP: u8 = b'\n';
+
+pub fn to_base64(s: Vec<u8>) -> Option<Bytes> {
```

Comment:
Can we change the `s` parameter as a `&[u8]` type? seems there is no need to accept it as a `Vec`

Reply:
ok, i changed to use a point as param